File: 2023-12-06-how-to-make-multiple-security-filter-chains/action-in-blog/src/main/java/action/in/blog/controller/UserController.java
package action.in.blog.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/app/api")
public class UserController {

    @GetMapping("/articles")
    public String articles() {
        return "user articles";
    }

    @GetMapping("/private-articles")
    public String privateArticles() {
        return "user's private articles";
    }

}

File: 2023-08-13-jpa-entity-serialize-exception-with-redis-session/action-in-blog/src/main/kotlin/action/in/blog/controller/UserController.kt
package action.`in`.blog.controller

import action.`in`.blog.service.UserService
import jakarta.servlet.http.HttpServletRequest
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(
    private val userService: UserService
) {

    private val sessionKeyUser: String = "SESSION_KEY_USER"

    @GetMapping("/users/{id}")
    fun users(
        servletRequest: HttpServletRequest,
        @PathVariable("id") id: Long
    ) {
        val user = userService.getUser(id)
        val session = servletRequest.getSession(true)
        session.setAttribute(sessionKeyUser, user)
    }
}